app/src/main/java/com/example/looperhandler/MainActivity.java

```diff
@@ -0,0 +1,175 @@
+package com.example.looperhandler;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.Message;
+import android.os.SystemClock;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.navigation.ui.AppBarConfiguration;
+
+import com.example.looperhandler.databinding.ActivityMainBinding;
+
+import java.lang.ref.WeakReference;
+
+public class MainActivity extends AppCompatActivity {
+
+    private AppBarConfiguration appBarConfiguration;
+    ActivityMainBinding binding;
+
+    private ExampleLooper looper = new ExampleLooper();
+    private ExampleHandlerThread handlerThread = new ExampleHandlerThread();
+//    private Handler handler;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        binding = ActivityMainBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        binding.button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                handlerThread.start();
+            }
+        });
+
+        binding.button3.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ExampleAsyncTask asyncTask = new ExampleAsyncTask(MainActivity.this);
+                asyncTask.execute(10);
+
+            }
+        });
+
+
+        binding.button2.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Message msg = Message.obtain(handlerThread.getHandler());
+                msg.what = 1;
+                msg.arg1 = 123;
+                msg.obj = "Obj String";
+
+                handlerThread.getHandler().sendMessage(msg);
+
+            }
+        });
```

Comment:
You can save yourself from having to override the `onClick` function multiple times by simply extending your Activity with `View.OnClickListener`. Learn how to do so [here](https://www.youtube.com/watch?v=GtxVILjLcw8&disableadblock=1).

Reply:
I was getting this warning and code wasnt working.
"Resource IDs will be non-final in Android Gradle Plugin version 5.0, avoid using them in switch case statements" 
So i used if/else and compared the views instead, not sure if that is the best practice.